### PR TITLE
[FIX] hr: singleton error on writing user to employee

### PR DESCRIPTION
**Pull request by niyasraphy**
before this commit, on writing user_id to hr.employee on multiple record in raisng single ton error.

after this commit, no error wont be raised on the
same.

Issue: https://github.com/odoo/odoo/issues/129620

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/hr/models/hr_employee.py [hunk 1]
```diff
@@ -305,19 +305,19 @@ def create(self, vals):
         return employee
 
     def write(self, vals):
-        if 'address_home_id' in vals:
-            account_id = vals.get('bank_account_id') or self.bank_account_id.id
-            if account_id:
-                self.env['res.partner.bank'].browse(account_id).partner_id = vals['address_home_id']
-            self.message_unsubscribe(self.address_home_id.ids)
-            if vals['address_home_id']:
+        for rec in self:
```
Comment (sofiagvaladze) [inline addons/hr/models/hr_employee.py:308]: Hey @niyasraphy :wave: 

Thanks for your fix ^^

Iterating through the recordset is really not a good idea in the write method. 
My colleague proposed another solution for it: https://gist.github.com/kba-odoo/4eba30d56348603f271849cf4fb0522e#file-diff-L25

What do you think? :smile:

Comment (niyasraphy) [reply addons/hr/models/hr_employee.py]: thanks @sofiagvaladze ,
indeed it was the easiest way, updated the PR

**Comment by sofiagvaladze** [comment]
@niyasraphy no need to remove that if check :smile: 

I was talking about something like this - https://github.com/odoo-dev/odoo/commit/37802e4dbf7aff41ce0766ce4d81b3f8cf96db9e

**Comment by niyasraphy** [comment]
ah, may bad, i missed that change of account .
thanks again @sofiagvaladze

**Comment by sofiagvaladze** [comment]
robodoo r+

**Pull request closed without merging**